### DISCUSSION TITLE
Discussion: What do you think about using Scintilla?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#233] [Ideas]
`eepp` is a C++ code base, so I think it's not hard to integrate Scintilla.

**Comment by SpartanJ** [comment]
Scintilla provides an interface to develop a full editor, but we already have all that ready and working, integrating it is almost impossible today. *But*, it's possible to integrate the lexers via [Lexilla](https://www.scintilla.org/Lexilla.html), the Scintilla lexer library. As for the moment I don't see any great advantage of adding an additional lexer, we already support plenty of languages and there's no particular advantage of including a secondary lexer. I do think that it sounds more interesting adding tree-sitter support given that it has really nice results for syntax-highlighting. Scintilla lexers are good but not *great*.

**Reply by ghost** [reply]
Thank you for the detailed answer. Btw, I don't think you can use Lexilla without Scintilla.

**Reply by SpartanJ** [reply]
It's possible! It's just not very convenient (the API is not very nice). When I started developing an editor I wanted to lean how all this worked, so I skipped Scintilla because it has all solved for you, it was not fun to use it from the learning point of view, I wasn't planning doing a full blown editor, just a very simple editor to integrate to the eepp GUI editor, but I ended up getting too enthusiastic about it and now we have ecode. I might had used it in a different situation though.
